pinecone/db_data/resources/sync/namespace.py

```diff
@@ -0,0 +1,120 @@
+from typing import Optional, Iterator
+
+from pinecone.core.openapi.db_data.api.namespace_operations_api import NamespaceOperationsApi
+from pinecone.core.openapi.db_data.models import (
+    ListNamespacesResponse,
+    NamespaceDescription,
+)
+
+from pinecone.utils import install_json_repr_override, PluginAware
+
+from .namespace_request_factory import NamespaceRequestFactory
+
+for m in [ListNamespacesResponse, NamespaceDescription]:
+    install_json_repr_override(m)
+
+
+class NamespaceResource(PluginAware):
+    def __init__(
+        self,
+        api_client,
+        config,
+        openapi_config,
+        pool_threads: int,
+    ) -> None:
+        self.config = config
+        """ :meta private: """
+
+        self._openapi_config = openapi_config
+        """ :meta private: """
+
+        self._pool_threads = pool_threads
+        """ :meta private: """
+
+        self.__namespace_operations_api = NamespaceOperationsApi(api_client)
+        super().__init__()
+
+    def describe(self, namespace: str) -> NamespaceDescription:
+        """
+        Args:
+            namespace (str): The namespace to describe
+
+        Returns:
+            `NamespaceDescription`: Information about the namespace including vector count
+
+        Describe a namespace within an index, showing the vector count within the namespace.
+        """
+        args = NamespaceRequestFactory.describe_namespace_args(namespace=namespace)
+        return self.__namespace_operations_api.describe_namespace(**args)
+
+    def delete(self, namespace: str):
+        """
+        Args:
+            namespace (str): The namespace to delete
+
+        Delete a namespace from an index.
+        """
+        args = NamespaceRequestFactory.delete_namespace_args(namespace=namespace)
+        return self.__namespace_operations_api.delete_namespace(**args)
+
+    def list(self, **kwargs) -> Iterator[ListNamespacesResponse]:
+        """
+        Args:
+            limit (Optional[int]): The maximum number of namespaces to fetch in each network call. If unspecified, the server will use a default value. [optional]
+            pagination_token (Optional[str]): When there are multiple pages of results, a pagination token is returned in the response. The token can be used
+                to fetch the next page of results. [optional]
+
+        Returns:
+            Returns a generator that yields each namespace. It automatically handles pagination tokens on your behalf so you can
+            easily iterate over all results. The `list` method accepts all of the same arguments as list_paginated
+
+        ```python
+        for namespace in index.list_namespaces():
+            print(namespace)
+        ```
```

Comment:
Docstrings now need to use RST syntax for sphinx to parse them correctly. Instead of triple bacticks, you need to use the `.. code-block:: python` directive. The example code needs to be indented one line from the indent level of the directive. And (very important) you have to have a blank line between the directive start and the first line of your code, otherwise nothing will be rendered. You can search `code-block` in the repo to find many examples

```suggestion
       .. code-block:: python
        
          for namespace in index.list_namespaces():
              print(namespace)
```